homeassistant/components/subaru/sensor.py

```diff
@@ -220,22 +220,18 @@ def native_value(self) -> None | int | float:
         if current_value is None:
             return None
 
-        if unit in LENGTH_UNITS:
+        if unit in {LENGTH_KILOMETERS, LENGTH_MILES}:
             return round(unit_system.length(current_value, unit), 1)
 
-        if unit in PRESSURE_UNITS and unit_system == IMPERIAL_SYSTEM:
+        if unit == PRESSURE_HPA and unit_system.pressure_unit == PRESSURE_PSI:
```

Comment:
This solution has the advantage that:
- it keeps a `positive assert`, so it won't need to be adjusted if/when new systems are introduced
- it doesn't need any adjustments to UnitSystem